typescript/src/entity.ts

```diff
@@ -457,6 +458,11 @@ export class Entity {
       }
       result.set(name, value);
     };
+    if ((this as any).fileData !== undefined) {
+      appendData(String(fileIndex), (this as any).fileData);
```

Comment:
Sorry, it this comment stupid.

it looks so so weird, base on my existing knowledge. sincePreviously serialise function will out put JSON result and files array pair.

now you can directly write files into JSON result, how does it works?
1. backend should not accept it right? since backend will looking for files from request.files instead request.values?
2. if fetch can add files to request by default, how could you guarantee the order is the same as fileIndex?

Reply:
> now you can directly write files into JSON result

`toFormData` doesn't generate JSON, it generates `FormData` object.

I think that answers your entire question.

> fetch can add files to request by default

do you refer to `RequestOptions`. I haven't done anything about that. It's not a unique to files. If the SDK user is stupid, and makes trouble by intentionally setting a wrong content-type, or sends any other wrong header or url.. yes, they can break things at the moment.